frontend/packages/policy-editor/src/components/ExpandablePolicyCard/ResourceNarrowingList/ResourceNarrowingList.test.tsx

```diff
@@ -0,0 +1,156 @@
+const mockResource2: PolicyRuleResource = { type: 'type2', id: 'id2' };
+const mockResources: PolicyRuleResource[] = [mockResource1, mockResource2];
+
+const mockUsageType: PolicyEditorUsage = 'app';
+
+const mockNewText: string = 'test';
+
+describe('ResourceNarrowingList', () => {
+  afterEach(jest.clearAllMocks);
+
+  const mockHandleInputChange = jest.fn();
+  const mockHandleRemoveResource = jest.fn();
+  const mockHandleClickAddResource = jest.fn();
+  const mockHandleRemoveElement = jest.fn();
+  const mockHandleCloneElement = jest.fn();
+  const mockOnBlur = jest.fn();
+
+  const defaultProps: ResourceNarrowingListProps = {
+    resources: mockResources,
+    handleInputChange: mockHandleInputChange,
+    handleRemoveResource: mockHandleRemoveResource,
+    handleClickAddResource: mockHandleClickAddResource,
+    handleRemoveElement: mockHandleRemoveElement,
+    handleCloneElement: mockHandleCloneElement,
+    onBlur: mockOnBlur,
+    usageType: mockUsageType,
+  };
+
+  it('renders the list of resources', () => {
+    render(<ResourceNarrowingList {...defaultProps} />);
+
+    const removeButtons = screen.getAllByRole('button', {
+      name: textMock('policy_editor.narrowing_list_field_delete'),
+    });
+    const typeInputs = screen.getAllByLabelText(
+      textMock('policy_editor.narrowing_list_field_type')
+    );
+    const idInputs = screen.getAllByLabelText(textMock('policy_editor.narrowing_list_field_id'));
+
+    const numItems: number = mockResources.length;
+    expect(removeButtons).toHaveLength(numItems);
+    expect(typeInputs).toHaveLength(numItems);
+    expect(idInputs).toHaveLength(numItems);
+
+    const addButton = screen.getByText(textMock('policy_editor.narrowing_list_add_button'));
+    expect(addButton).toBeInTheDocument();
+  });
+
+  it('does not show the delete button for the first resource when "usageType" is resource', () => {
+    render(<ResourceNarrowingList {...defaultProps} usageType='resource' />);
+
+    const removeButtons = screen.getAllByRole('button', {
+      name: textMock('policy_editor.narrowing_list_field_delete'),
+    });
+    const numItems: number = mockResources.length;
+    expect(removeButtons).toHaveLength(numItems - 1); // Minus first element which is readonly
+  });
+
+  it('calls "handleInputChange" when id or type is edited', async () => {
+    const user = userEvent.setup();
+    render(<ResourceNarrowingList {...defaultProps} />);
+
+    const [idInput] = screen.getAllByLabelText(textMock('policy_editor.narrowing_list_field_id'));
+    await act(() => user.type(idInput, mockNewText));
+
+    const [typeInput] = screen.getAllByLabelText(
+      textMock('policy_editor.narrowing_list_field_type')
+    );
+    await act(() => user.type(typeInput, mockNewText));
+
+    const numberOfTimes: number = mockNewText.length * 2; // Length of word for id field and length of word for type field
+    expect(mockHandleInputChange).toHaveBeenCalledTimes(numberOfTimes);
+  });
```

Comment:
Her kan du godt gjøre testen mer målrettet ved å sjekke kallhistorikken etter hver endring. Du kan bruke `mockHandleInputChange.mockClear()` for å tilbakestille historikken mellom sjekkene. Det jeg tenker på er noe slikt:
```suggestion
  it('calls "handleInputChange" when id or type is edited', async () => {
    const user = userEvent.setup();
    render(<ResourceNarrowingList {...defaultProps} />);

    const [idInput] = screen.getAllByLabelText(textMock('policy_editor.narrowing_list_field_id'));
    await act(() => user.type(idInput, mockNewText));
    expect(mockHandleInputChange).toHaveBeenCalledTimes(mockNewText.length);

    mockHandleInputChange.mockClear();

    const [typeInput] = screen.getAllByLabelText(
      textMock('policy_editor.narrowing_list_field_type')
    );
    await act(() => user.type(typeInput, mockNewText));
    expect(mockHandleInputChange).toHaveBeenCalledTimes(mockNewText.length);
  });
```

Reply:
I agree, thanks!